test/evp_kdf_test.c

```diff
@@ -239,6 +239,41 @@ static int test_kdf_hkdf(void)
     return ret;
 }
 
+static int kdf_hkdf_fixed_digest(const char *kdf_name, char *digest)
```

Comment:
I don't think this is the best place for this test.  I'd suggest adding it to `evp_test` by adding cases to the data files in `test/recipes/30-test_evp_data/`.

A test here would be worthwhile if you are adding a negative test to verify the change of digest behaviour.  That depends on how a digest change is being handled.

Reply:
I will move this consistency test to `test/recipes/30-test_evp_data/` as suggested.

The negative test is below in kdf_hkdf_fixed_digest_change_digest. With the more user-friendly change above, I will expand this test.